pkg/webui/console/components/webhook-form/index.js

```diff
@@ -165,6 +169,96 @@ const validationSchema = Yup.object().shape({
     .test('has path length at most 64 characters', m.messagePathValidateTooLong, messageCheck),
 })
 
+// Encode and decode basic auth header.
+let currentHeaders
+
+const mapBasicAuthHeaderToBoolean = value => {
+  currentHeaders = value
+  const useBasicAuth =
+    value?.some(header => header.key === 'Authorization') &&
+    value?.some(header => header.value.startsWith('Basic'))
+  return useBasicAuth
+}
+
+const mapBooleanToBasicAuthHeader = value => {
+  if (value) {
+    return [...currentHeaders, ...{ key: 'Authorization', value: 'Basic' }]
+  }
+  return currentHeaders.filter(
+    header => header.key !== 'Authorizarion' && !header.value.startsWith('Basic'),
+  )
+}
+
+const mapBasicAuthHeaderToUsername = value => {
+  const basicAuth = value.filter(
+    header => header.key === 'Authorization' && header.value.startsWith('Basic'),
+  )
+  const encodedCredentials = basicAuth[0]?.value.split('Basic')[1]
+  if (encodedCredentials) {
+    const decodedCredentials = atob(encodedCredentials)
+    const decodedUsername = decodedCredentials.slice(0, decodedCredentials.indexOf(':'))
+    return decodedUsername
+  }
+
+  return ''
+}
+
+const mapCredentialsToAuthHeader = (forUsername, fieldValue) => {
```

Comment:
```suggestion
const createCredentialsToAuthHeaderMapper = forUsername => fieldValue => {
```

See my other comment below.